### PR TITLE
Add Purpur Egg

### PR DESCRIPTION
**Pull request by Pemigrade**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

### New Server Submissions:

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your server use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the server to the main README.md?
4. [x] Have you added a unique README.md for the server you are adding?

### Changes to an existing Egg:

1. [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
2. [ ] Have you tested your Egg changes?

### Summary
A fork of Tuinity & Paper. Adds some new config options. Read https://github.com/pl3xgaming/purpur for more information.

### CODE REVIEW

**Comment by tmunsch** [comment]
You can remove the line that pulls a server.properties file from github as it's created with the proper settings on start. Beyond that the egg works fine.

**Comment by parkervcp** [comment]
> 
> 
> You can remove the line that pulls a server.properties file from github as it's created with the proper settings on start. Beyond that the egg works fine.

The reason we pull the server.properties is so the panel can set the port properly on first start.

File: minecraft/java/purpur/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+See https://github.com/pl3xgaming/purpur for additional information. I copy-pasted this readme from there.
```
Comment (parkervcp) [inline minecraft/java/purpur/README.md:1]: I don't need their entire README in my repo. Just a link to theirs is fine.

File: README.md [hunk 2]
```diff
@@ -100,6 +100,7 @@ If you are reading this it looks like you are looking to add an egg to your serv
   * [Forge](/minecraft/java/forge/)  
   * [Magma](/minecraft/java/magma/)
   * [Paper](/minecraft/java/paper)
+  * [Purpur](/minecraft/java/purpur)
```
Comment (parkervcp) [inline README.md:103]: Please rebase on the current master

Comment (Pemigrade) [reply README.md:110]: Should be fixed

File: minecraft/java/purpur/README.md (hunk 1 above)
Comment (Pemigrade) [reply minecraft/java/purpur/README.md]: Done!

**Comment by Pemigrade** [comment]
^ most recent commit contains large changes that make the egg use purpur's API rather than their CI server. Also adds support for installing with version and build. Tested the new egg and all variables work

File: minecraft/java/purpur/egg-purpur.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
.../${MINECRAFT_VERSION} | jq -r --arg BUILD ${BUILD_NUMBER} '.builds.all[] | contains($BUILD)' | grep true`\r\nLATEST_PURPUR_BUILD=`curl -s https:\/\/purpur.pl3x.net\/api\/v1\/purpur\/${MINECRAFT_VERSION} | jq -r '.builds.latest'`\r\n\r\nif [ \"${BUILD_EXISTS}\" == \"true\" ] || [ ${BUILD_NUMBER} == \"latest\" ]; then\r\n    echo -e \"Build is valid. Using version ${BUILD_NUMBER}\"\r\nelse\r\n    echo -e \"Using the latest Purpur build\"\r\n    BUILD_NUMBER=${LATEST_PURPUR_BUILD}\r\nfi\r\n\r\necho \"Version being downloaded\"\r\necho -e \"MC Version: ${MINECRAFT_VERSION}\"\r\necho -e \"Build: ${BUILD_NUMBER}\"\r\nDOWNLOAD_URL=https:\/\/purpur.pl3x.net\/api\/v1\/purpur\/${MINECRAFT_VERSION}\/${BUILD_NUMBER}\/download \r\n\r\n\r\ncd \/mnt\/server\r\n\r\necho -e \"running curl -o ${SERVER_JARFILE} ${DOWNLOAD_URL}\"\r\n\r\nif [ -f ${SERVER_JARFILE} ]; then\r\n    mv ${SERVER_JARFILE} ${SERVER_JARFILE}.old\r\nfi\r\n\r\ncurl -o ${SERVER_JARFILE} ${DOWNLOAD_URL}\r\n\r\nif [ ! -f server.properties ]; then\r\n    echo -e \"Downloading MC server.properties\"\r\n    curl -o server.properties https:\/\/raw.githubusercontent.com\/parkervcp\/eggs\/master\/minecraft\/java\/server.properties\r\nfi",
```
Comment (parkervcp) [inline minecraft/java/purpur/egg-purpur.json:23]: You can add a `-sSL` to the curl on line 47 of the install script to make it somewhat more resilient.

Comment (Pemigrade) [reply minecraft/java/purpur/egg-purpur.json]: Done! Thanks for the tip.

**Pull request merged**